src/pages/profile/index.tsx

```diff
@@ -39,14 +39,35 @@ profile.get<"/:handle">(async (c) => {
   const contStr = c.req.query("cont");
   const cont = contStr == null || contStr.trim() === "" ? undefined : contStr;
   if (cont != null && !isUuid(cont)) return c.notFound();
+  const pageStr = c.req.query("page");
+  if (
+    pageStr !== undefined &&
+    (Number.isNaN(Number.parseInt(pageStr)) || Number.parseInt(pageStr) < 1)
+  ) {
+    return c.notFound();
+  }
+  const page =
+    pageStr !== undefined && !Number.isNaN(Number.parseInt(pageStr))
+      ? Number.parseInt(pageStr)
+      : 1;
+  const totalPosts = await db.query.posts.findMany({
+    where: and(
+      eq(posts.accountId, owner.id),
+      or(eq(posts.visibility, "public"), eq(posts.visibility, "unlisted")),
+    ),
+  });
+  const maxPage = Math.ceil(totalPosts.length / PAGE_SIZE);
```

Comment:
This code retrieves all posts that meets the criteria just for counting the posts. A more efficient way to count the rows without retrieving all the data would be to use a [`count()` query](https://orm.drizzle.team/docs/select#aggregations-helpers) directly:

```suggestion
  const [{ totalPosts }] = await db
    .select({ totalPosts: count() })
    .from(posts)
    .where(
      and(
        eq(posts.accountId, owner.id),
        or(eq(posts.visibility, "public"), eq(posts.visibility, "unlisted")),
      )
    );
  const maxPage = Math.ceil(totalPosts / PAGE_SIZE);
```

Of course, you need to import `count` from `"drizzle-orm"` module:

```typescript
import { and, count, desc, eq, or } from "drizzle-orm";
```

Reply:
thank you! I fixed it. (https://github.com/fedify-dev/hollo/pull/105/commits/ba8e0de601104e5c1b089618af08f11bbca1247f)